src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -197,41 +197,31 @@ export const txHistoryApi = createApi({
   reducerPath: 'txHistoryApi',
   endpoints: build => ({
     getFoxyRebaseHistoryByAccountId: build.query<RebaseHistory[], RebaseTxHistoryArgs>({
-      queryFn: async ({ accountId, portfolioAssetIds }, { dispatch }) => {
+      queryFn: ({ accountId, portfolioAssetIds }, { dispatch }) => {
         const { chainId, account: userAddress } = fromAccountId(accountId)
         // foxy is only on eth mainnet, and [] is a valid return type and won't upsert anything
         if (chainId !== ethChainId) return { data: [] }
         // foxy contract address, note not assetIds
-        const foxyTokenContractAddressWithBalances = foxyAddresses.reduce<string[]>(
-          (acc, { foxy }) => {
-            const contractAddress = foxy.toLowerCase()
-            portfolioAssetIds.some(id => id.includes(contractAddress)) && acc.push(contractAddress)
-            return acc
-          },
-          [],
-        )
+        const foxyTokenContractAddress = (() => {
+          const contractAddress = foxyAddresses[0].foxy.toLowerCase()
+          if (portfolioAssetIds.some(id => id.includes(contractAddress))) return contractAddress
+        })()
 
-        // don't do anything below if we don't hold a version of foxy
-        if (!foxyTokenContractAddressWithBalances.length) return { data: [] }
+        // don't do anything below if we don't have FOXy as a portfolio AssetId
+        if (!foxyTokenContractAddress) return { data: [] }
 
         // setup foxy api
         const foxyApi = getFoxyApi()
 
-        await Promise.all(
-          foxyTokenContractAddressWithBalances.map(async tokenContractAddress => {
-            const rebaseHistoryArgs = { userAddress, tokenContractAddress }
-            const data = await foxyApi.getRebaseHistory(rebaseHistoryArgs)
-            const assetReference = tokenContractAddress
-            const assetNamespace = 'erc20'
-            const assetId = toAssetId({ chainId, assetNamespace, assetReference })
-            const upsertPayload = { accountId, assetId, data }
-            if (data.length) dispatch(txHistory.actions.upsertRebaseHistory(upsertPayload))
-          }),
-        )
-
-        // we don't really care about the caching of this, we're dispatching
-        // into another part of the portfolio above, we kind of abuse RTK query,
-        // and we're always force refetching these anyway
```

Comment:
Comment doesn't make sense anymore - this is now an IIAFE and acts as a comment that there's no data we're depending on since this isn't blocking